### ISSUE TITLE
switching Javascript-execution context to different iframe

### ISSUE THREAD

**Issue by stuart-little** [SeleniumHQ/selenium#8663]
## Switching Javascript-execution context to a (possibly cross-origin) iframe embedded in the main frame.

I would like to emulate Javascript-executing-context changes, as achievable

- in [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/console/reference#context) through a drop-down menu;
- in the Firefox Web Console, similarly (drop-down menu in the upper-right console corner);
- in the Firefox Web Console through the `cd(iframe)` [facility](https://bugzilla.mozilla.org/show_bug.cgi?id=1605327).

## Motivation

Automating logins that embed a cross-origin iframe for 2fa, for instance.

## Example

I have a login window as described above, that embeds a prompt requesting an additional passcode. I have tried:

(1) Switching context "manually" to the iframe in the console (both Firefox and Chrome); I can then access the relevant buttons and forms through selectors, etc.

(2) Locating the iframe with Firefox headless + `node` + `selenium-webdriver` and switching to it:

```js
ifr = driver.findElement(By.id('frame_id'));
driver.switchTo().frame(ifr);
```
This does switch and gives me access to a `DOM`, but it is *not* the one I can see in the console/inspector. In particular there are no buttons to access, etc.

In short: the `switchTo(iframe)` function does not seem to mirror Firefox' `cd(iframe)`.
